account_invoice_refund_reason/wizard/account_move_reversal.py

```diff
@@ -19,6 +19,22 @@ def _compute_reason(self):
                 record.reason = record.reason_id.name
 
     def reverse_moves(self):
+        """Overriden to set the reason_id fields in the new created refunds"""
         res = super().reverse_moves()
-        self.move_ids.mapped("reversal_move_id").write({"reason_id": self.reason_id.id})
```

Comment:
I think it's simpler to do:

```suggestion
        self.move_ids.mapped("reversal_move_id").filtered(lambda x: not x.reason_id).write({"reason_id": self.reason_id.id})
```